### PR TITLE
Displaying posts

### PR DESCRIPTION
**Pull request by dawidzon**
I would like to ask you to check my homework. I would like to practice doing pull requests also.

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -26,10 +26,9 @@ a:hover {
 
 body {
   margin: 0;
-  display: flex;
-  place-items: center;
-  min-width: 320px;
-  min-height: 100vh;
+
+  /* min-width: 320px;
+  min-height: 100vh; */
```
Comment (wozny1989) [inline src/index.css]: ✂️

File: src/pages/PostsList/components/PostPreview.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+const IconText = ({ icon, text }) => (
+  <Space>
+    {React.createElement(icon)}
+    {text}
+  </Space>
+)
+
+const PostPreview = ({ post }) => {
+  console.log(typeof post.body)
```
Comment (wozny1989) [inline src/pages/PostsList/components/PostPreview.jsx]: ✂️

File: src/pages/PostsList/components/PostPreview.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    let counter = 0
+    for (let i = 0; i < text.length; i++) {
+      if (text[i] === '.') {
+        counter++
+        if (counter === 3) {
+          return i + 1
+        }
+      }
+    }
+  }
```
Comment (wozny1989) [inline src/pages/PostsList/components/PostPreview.jsx]: Nice! Good for practice but quite hard to understand. 
How about:
```suggestion
  function getPositionThirdDot(string) {
    const char = '.'
    const index = 3;
    return string.split(char, index).join(char).length;
  }
```

**Review by wozny1989 — CHANGES_REQUESTED** [review]
Nice! Only few comments 👌

**Comment by dawidzon** [comment]
Thank you. I improved the display of posts as suggested by Marzena on Slack.

File: src/pages/PostsList/components/PostPreview.jsx [hunk 4]
```diff
@@ -37,7 +25,16 @@ const PostPreview = ({ post }) => {
...
-        description={post.body.slice(0, countToThreeDots(post.body))}
+        description={
+          <Paragraph
+            ellipsis={{
+              rows: 2,
+              expandable: true,
+            }}
+          >
+            {post.body}
+          </Paragraph>
```
Comment (wozny1989) [inline src/pages/PostsList/components/PostPreview.jsx]: Even better! ❤️

**Review by wozny1989 — APPROVED** [review]
Approve ✅ but remember to remove `console.log` here https://github.com/dawidzon/React-Blog/pull/1/files/6589ef879114518291a1882629959bbff56a011f#r1103752818 😄

File: db.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (moraw9) [inline db.json]: Do you need this file?

File: package.json [hunk 6]
```diff
@@ -6,20 +6,38 @@
   "scripts": {
     "dev": "vite",
     "build": "vite build",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "server": "json-server --watch posts.json --delay 3000"
   },
   "dependencies": {
+    "@ant-design/icons": "^5.0.1",
```
Comment (moraw9) [inline package.json]: Please remove `^` and re-install packages

File: src/App.jsx [hunk 7]
```diff
@@ -1,9 +1,12 @@
-import { PostsList } from './pages/PostsList.jsx'
 import './App.css'
+
 import { useState } from 'react'
 
-function App() {
+import { PostsList } from './pages'
+
+const App = () => {
   const [state, setState] = useState(0)
```
Comment (moraw9) [inline src/App.jsx]: You can remove it

File: src/pages/PostsList/components/PostPreview.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import React from 'react'
+
+const { Paragraph } = Typography
+
+const IconText = ({ icon, text }) => (
+  <Space>
+    {React.createElement(icon)}
+    {text}
+  </Space>
+)
```
Comment (moraw9) [inline src/pages/PostsList/components/PostPreview.jsx]: Please move it to separate file

File: src/pages/PostsList/components/PostPreview.jsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+const PostPreview = ({ post }) => {
+  return (
```
Comment (moraw9) [inline src/pages/PostsList/components/PostPreview.jsx]: If you only return something inside component, you can use this pattern:
```suggestion
const PostPreview = ({ post }) => <List>... </List>
```
So without curly brackets and `return` statement

File: src/pages/PostsList/components/PostPreview.jsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        <Button key="button-vertical">More...</Button>,
+      ]}
+    >
+      <List.Item.Meta
+        title={post.title}
+        description={
+          <Paragraph
+            ellipsis={{
+              rows: 2,
+              expandable: true,
```
Comment (moraw9) [inline src/pages/PostsList/components/PostPreview.jsx]: We will not allow to expand text, we'll add navigation to page with full content, so please remove `expandable: true`

File: src/index.css [hunk 11]
```diff
@@ -6,7 +6,7 @@
 
   color-scheme: light dark;
   color: rgba(255, 255, 255, 0.87);
-  background-color: #242424;
```
Comment (moraw9) [inline src/index.css:9]: You can remove all styles and only import `antd` styles `@import "antd/dist/reset.css";`

**Review by moraw9 — CHANGES_REQUESTED** [review]

File: src/pages/PostsList/components/PostPreview.jsx (hunk 8 above)
Comment (dawidzon) [reply src/pages/PostsList/components/PostPreview.jsx]: should it be a separate component?

Comment (moraw9) [reply src/pages/PostsList/components/PostPreview.jsx]: Yes, overall we avoid to declare more than one component inside one file.

**Comment by dawidzon** [comment]
Thank you for the tips. Please check the next homework.

File: src/App.css [hunk 12]
```diff
@@ -1,4 +1,4 @@
-#root {
+/* #root {
```
Comment (moraw9) [inline src/App.css]: If it's commented, you can remove it?

File: src/App.jsx [hunk 13]
```diff
@@ -1,10 +1,22 @@
...
-function App() {
-  const [state, setState] = useState(0)
-  return <PostsList />
+import { Navigate, Route, Routes } from 'react-router-dom'
+
+import { MainLayout } from './components'
+import { PostDetails, PostsList } from './pages'
+
+const App = () => {
+  return (
```
Comment (moraw9) [inline src/App.jsx]:
```suggestion
const App = () =>  (
```

File: src/pages/PostDetails/PostDetails.jsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+import { Loading } from '../../components'
+import { IconText } from '../PostsList/components'
+
+const PostDetails = () => {
+  const { postId } = useParams()
+  const [post, setPost] = useState()
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    setLoading(true)
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: ✂️

File: src/pages/PostDetails/PostDetails.jsx [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+      })
+      .finally(() => setLoading(false))
+  }, [])
+
+  return (
+    <>
+      {loading && <Loading />}
+      {!loading && (
+        <>
+          <h1>{post && post.title}</h1>
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: You can remove `post &&` due to initial loading value is true

File: src/pages/PostsList/PostsList.jsx [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import { List } from 'antd'
+import { useEffect, useState } from 'react'
+
+import { Loading } from '../../components'
+import { PostPreview } from './components'
```
Comment (moraw9) [inline src/pages/PostsList/PostsList.jsx]: I'll check issue with imports

**Review by moraw9 — CHANGES_REQUESTED** [review]
Looks good! 🚀  Only small changes to add

**Comment by dawidzon** [comment]
Thank you for checking. I made corrections

File: src/App.jsx [hunk 17]
```diff
@@ -1,10 +1,20 @@
...
 import './App.css'
-import { useState } from 'react'
 
-function App() {
-  const [state, setState] = useState(0)
-  return <PostsList />
-}
+import { Navigate, Route, Routes } from 'react-router-dom'
+
+import { MainLayout } from './components'
```
Comment (moraw9) [inline src/App.jsx]:
```suggestion
import { MainLayout } from 'components'
```

File: src/pages/PostDetails/PostDetails.jsx [hunk 18]
```diff
@@ -0,0 +1,35 @@
+import { LikeOutlined } from '@ant-design/icons'
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import { Loading } from '../../components'
+import { IconText } from '../PostsList/components'
```
Comment (moraw9) [inline src/pages/PostDetails/PostDetails.jsx]: Fix imports. You import `IconText` from `PostsList` so, you can move `IconText` to global `components` folder

File: src/pages/PostsList/PostsList.jsx (hunk 16 above)
Comment (moraw9) [reply src/pages/PostsList/PostsList.jsx]: Try to import them using auto import suggestion (absolute path for Loading)

**Review by moraw9 — CHANGES_REQUESTED** [review]
Great, only small changes with imports :D

**Comment by dawidzon** [comment]
I checked all components for imports. Now it should match.

**Review by moraw9 — APPROVED** [review]
Great, merge! ❤️

**Pull request merged**